pkg/scaler/prober.go

```diff
@@ -156,8 +158,16 @@ func (p *prober) tryAndRun(prepareRun func() (stopCh <-chan struct{})) error {
 	internalClient, internalSHA, internalErr := p.getClientFromSecret(p.probeDeps.Probe.Internal.KubeconfigSecretName, p.internalSHA)
 	externalClient, externalSHA, externalErr := p.getClientFromSecret(p.probeDeps.Probe.External.KubeconfigSecretName, p.externalSHA)
 
-	// Run a fresh prober goroutine if any one of the secrets changed
-	if internalErr == nil || externalErr == nil {
+	if apierrors.IsNotFound(internalErr) || apierrors.IsNotFound(externalErr) {
```

Comment:
I was thinking over the week-end if we should change it to the following to cancel the probe if any error other than `AlreadyExists` occurs.
```suggestion
	if !apierrors.IsAlreadyExists(internalErr) && !apierrors.IsAlreadyExists(externalErr) {
```

Reply:
Also, shouln't we have a `shootWokeUp` or `shootInHibernationTransition` check here? Please note that this code-path might be reached from the `secretsInformers` event as well.

Reply:
> if we should change it to the following to cancel the probe if any error other than AlreadyExists occurs

hmm...makes sense. So if there is a transient network error, would the probe start again for the namespace when the error goes away (even if there are no updates to secrets or cluster state)?

>shouln't we have a shootWokeUp or shootInHibernationTransition check here?

in `probe()` we have already added check to close a probe if shoot is not up and ready. would we still need such a check here to close a probe? wdyt?